Python/executor.c

```diff
@@ -30,6 +30,19 @@
 #undef ENABLE_SPECIALIZATION
 #define ENABLE_SPECIALIZATION 0
 
+#undef DISPATCH_INLINED
+#define DISPATCH_INLINED(NEW_FRAME)                             \
+    do {                                                        \
+        assert(tstate->interp->eval_frame == NULL);             \
+        _PyFrame_SetStackPointer(frame, stack_pointer);         \
+        frame->prev_instr -= 1;                                 \
```

Comment:
Why isn't this `next_instr - 1` anymore?

Reply:
Because Tier 2 doesn't have a `next_instr` variable. That's also why we have to sneak in the `SAVE_IP` uop -- it sets `prev_instr`. The `_PUSH_FRAME` uop (and hence this macro) literally doesn't have enough information to know what to set it to -- the oparg provided to `SAVE_IP` is calculated by the translator (and I don't want to abuse `operand` or do other irregular things here).